controllers/install/lookoutingester_controller.go

```diff
@@ -0,0 +1,267 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	installv1alpha1 "github.com/armadaproject/armada-operator/apis/install/v1alpha1"
+	"github.com/armadaproject/armada-operator/controllers/builders"
+)
+
+// TODO: Pretty sure all these constants need re-visiting.
+const (
+	lookoutIngesterFinalizer = "batch.tutorial.kubebuilder.io/finalizer" // TODO(Clif): Do we need finalizers?
+)
+
+// LookoutIngesterReconciler reconciles a LookoutIngester object
+type LookoutIngesterReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=install.armadaproject.io,resources=lookoutingesters,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=install.armadaproject.io,resources=lookoutingesters/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=install.armadaproject.io,resources=lookoutingesters/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the LookoutIngester object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *LookoutIngesterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	started := time.Now()
+	logger.Info("Reconciling LookoutIngester object")
+
+	logger.Info("Fetching LookoutIngester object from cache")
+	var lookoutIngester installv1alpha1.LookoutIngester
+	if err := r.Client.Get(ctx, req.NamespacedName, &lookoutIngester); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("LookoutIngester not found in cache, ending reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	components, err := r.generateInstallComponents(&lookoutIngester)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	deletionTimestamp := lookoutIngester.ObjectMeta.DeletionTimestamp
+	// examine DeletionTimestamp to determine if object is under deletion
+	// TODO(Clif): Do we need a finalizer?
```

Comment:
As I said earlier, you don't need it.

Reply:
Addressed in 6c2f5f8737e28e2dd4b046a27c46e8b8084f9879